tools/release/backport_pr.sh

```diff
@@ -24,59 +24,100 @@ ensure_command () {
+  echo "     incorporate any generated files. Try running `git clean -xdff` beforehand." >/dev/stderr
+  echo "" >/dev/stderr
+  echo "Example: $0 25456 gnossen \"v1.30.x v1.31.x v1.32.x v1.33.x v1.34.x v1.35.x v1.36.x\" \"menghanl,gnossen\"" >/dev/stderr
+  echo "Example: $0 25493 gnossen \"\$(seq 30 33 | xargs -n1 printf \"v1.%s.x \")\" \"menghanl\" -c ./tools/dockerfile/push_testing_images.sh" >/dev/stderr
+  exit 1
+}
+
 ensure_command "curl"
 ensure_command "egrep"
 ensure_command "hub"
 ensure_command "jq"
 
+if [ "$#" -lt "4" ]; then
+  display_usage
+fi
+
+PR_NUMBER="$1"
+GITHUB_USER="$2"
+BACKPORT_BRANCHES="$3"
+REVIEWERS="$4"
+shift 4
+
+PER_BACKPORT_COMMAND=""
+while getopts "c:" OPT; do
+  case "$OPT" in
+    c )
+      PER_BACKPORT_COMMAND="$OPTARG"
+      ;;
+    \? )
+      echo "Invalid option: $OPTARG" >/dev/stderr
+      display_usage
+      ;;
+    : )
+      echo "Invalid option: $OPTARG requires an argument." >/dev/stderr
+      display_usage
+      ;;
+  esac
+done
+
+if [[ ! -z "$(git status --porcelain)" && ! -z "$PER_BACKPORT_COMMAND" ]]; then
+  echo "Your working directory is not clean. Try running `git clean -xdff`. Warning: This is irreversible." > /dev/stderr
+  exit 1
+fi
+
 if [ -z "$GITHUB_TOKEN" ]; then
   echo "A GitHub token is required to run this script. See " \
          "https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token" \
          " for more information" >/dev/stderr
   exit 1
 fi
 
-if [ "$#" != "4" ]; then
-  echo "USAGE: $0 PR_NUMBER GITHUB_USER BACKPORT_BRANCHES REVIEWERS" >/dev/stderr
-  echo "   PR_NUMBER: The number for the PR to be backported." >/dev/stderr
-  echo "   GITHUB_USER: Your GitHub username." >/dev/stderr
-  echo "   BACKPORT_BRANCHES: A space-separated list of branches to which to backport." >/dev/stderr
-  echo "   REVIEWERS: A comma-separated list of users add as reviewers and assignees." >/dev/stderr
-  echo "" >/dev/stderr
-  echo "Example: $0 25456 gnossen \"v1.30.x v1.31.x v1.32.x v1.33.x v1.34.x v1.35.x v1.36.x\" \"menghanl,gnossen\""
-  exit 1
-fi
-
-echo "This script will create a collection of backport PRs. Make sure the PR to" \
-       " backport has already been merged. You will probably " \
-       " have to touch your gnubby a frustrating number of times. C'est la vie."
+echo "This script will create a collection of backport PRs. You will probably " \
+       "have to touch your gnubby a frustrating number of times. C'est la vie."
 printf "Press any key to continue."
 read -r RESPONSE </dev/tty
 printf "\n"
 
-PR_NUMBER="$1"
-GITHUB_USER="$2"
-
-BACKPORT_BRANCHES="$3"
-
-REVIEWERS="$4"
 
 PR_DATA=$(curl -s -u "$GITHUB_USER:$GITHUB_TOKEN" \
           -H "Accept: application/vnd.github.v3+json" \
           "https://api.github.com/repos/grpc/grpc/pulls/$PR_NUMBER")
 
-MERGE_COMMIT=$(echo "$PR_DATA" | jq -r '.merge_commit_sha')
+STATE=$(echo "$PR_DATA" | jq -r '.state')
+if [ "$STATE" == "merged" ]; then
```

Comment:
When is the state attribute set to "merged"? It looks like GitHub lets you filter on state being "open", "closed", or "all" in the PR list API. https://docs.github.com/en/rest/reference/pulls#list-pull-requests

Reply:
Great catch! I should have gone back and checked on this. This would have broken the merged case. This made me realize that I didn't actually test out the old "already merged" case.

I've gone back and fixed a couple of other problems with that case. It should be good now.